### ISSUE TITLE
Switch of syntax check has no effect

### ISSUE THREAD

**Issue by Alisis33** [radimitrov/CSharpShellApp#165]
To switch of syntax check in options has no effect.

I hope it saves battery live.

**Comment by radimitrov** [comment]
I'll fix that. And I'll extend the range of the keystrokes thing.
Plus add another trigger method for diagnostics. 

I have no idea how much battery can be saved from turning it off.
The compiler does a mini benchmark just because when you click "compile". 
That is about 50ms on for an average Xamarin.Forms app on my device.

However the semantic model check (live compile time errors) doesn't generate IL and is limited to diagnostics for a single file. I haven't benchmarked it but I suspect it will multiple times less than 50ms. And the syntax check is even faster.
So shouldn't be any real battery drain if you leave it on a "lower frequency"

**Comment by Alisis33** [comment]
OK, I only hope... the IDE then simply consumes much...

**Comment by radimitrov** [comment]
Yeah. It's not just a text editor after all. After initialization it should be pretty well optimized, though.
However if you turn off a few things it should improve battery life a bit.
